main.qml

```diff
@@ -496,16 +495,30 @@ ApplicationWindow {
     }
 
     function onDeviceButtonRequest(code){
-        prevSplashText = splash.messageText;
-        splashDisplayedBeforeButtonRequest = splash.visible;
-        appWindow.showProcessingSplash(qsTr("Please proceed to the device..."));
+        if (txConfirmationPopup.visible) {
+            txConfirmationPopup.bottomTextAnimation.running = true
+            if (!txConfirmationPopup.errorText.visible) {
+                txConfirmationPopup.bottomText.text  = qsTr("Please confirm transaction in the device...") + translationManager.emptyString;
+            } else {
+                txConfirmationPopup.bottomText.text  = qsTr("Please proceed to the device...") + translationManager.emptyString;
+            }
+        } else {
+            prevSplashText = splash.messageText;
+            splashDisplayedBeforeButtonRequest = splash.visible;
+            appWindow.showProcessingSplash(qsTr("Please proceed to the device..."));
+        }
     }
 
     function onDeviceButtonPressed(){
-        if (splashDisplayedBeforeButtonRequest){
-           appWindow.showProcessingSplash(prevSplashText);
+        if (txConfirmationPopup.visible) {
+            txConfirmationPopup.bottomTextAnimation.running = false
+            txConfirmationPopup.bottomText.text  = qsTr("Signing transaction in the device...") + translationManager.emptyString;
         } else {
-           hideProcessingSplash();
+            if (splashDisplayedBeforeButtonRequest){
+                appWindow.showProcessingSplash(prevSplashText);
+            } else {
+                hideProcessingSplash();
+            }
         }
     }
```

Comment:
There is still a bug when trezor asks the user to authorize refreshing keys (which enters onDeviceButtonRequest()) while the txConfirmationPopup is visible. After the users press the device button, the device refreshes it keys, but the previous state of the txConfirmationPopup is not recovered (if the txConfirmationpopup state was default and cancel and confirm buttons were appearing, now the bottom text "Signing transaction in the device..." appears, but the device is not signing anything).

I don't know how to reproduce this refresh keys request in Trezor or what is the best approach to solve this.